lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2828,7 +2830,7 @@ public function createEntity($className, array $data, &$hints = [])
      */
     public function triggerEagerLoads()
     {
-        if ( ! $this->eagerLoadingEntities) {
+        if (empty($this->eagerLoadingEntities)) {
```

Comment:
This should be reverted, since a simple check on the array is sufficient, while `empty()` introduces further risks instead. Overall, avoid using `empty()`, as it is one of the many weird PHP operator-non-operator-will-work-with-too-many-things.

Reply:
I've also just added that because of scrutinizer complaining about it https://scrutinizer-ci.com/g/doctrine/doctrine2/inspections/2976232d-f9dc-4542-8680-7b6cb411aab7/issues/files/lib/Doctrine/ORM/UnitOfWork.php?status=new&orderField=path&order=asc&honorSelectedPaths=0